src/DataObject/Data/Adapter/VideoAdapter.php

```diff
@@ -67,9 +67,10 @@ public function getDataForSetter(
         }
 
         $type = $adapterData['type'] ?? null;
+        $poster = $adapterData['poster'] ?? null;
         if ($type === ElementTypes::TYPE_ASSET) {
             $adapterData['data'] = $this->resolveAssetIfNeeded($type, $adapterData['data']['fullPath']);
-            $adapterData['poster'] = $this->getAssetByPath($adapterData['poster'] ?? null);
+            $adapterData['poster'] = $this->getAssetByPath($poster ? $poster['fullPath'] : null);
```

Comment:
Not a massive problem, but shouldn't we resolve the asset relation for `poster` and `data` by ID instead of the path? If the asset gets moved in the meantime the ID would still work but the path would be broken...